frontend/src/employee-frontend/components/document-templates/template-editor/TemplateModal.tsx

```diff
@@ -104,18 +119,28 @@ export const documentTemplateForm = transformed(
 
     const output: DocumentTemplateBasicsRequest = {
       ...value,
-      processDefinitionNumber: archived
-        ? value.processDefinitionNumber.trim()
-        : null,
-      archiveDurationMonths: archived
-        ? parseInt(value.archiveDurationMonths)
-        : null,
+
       confidentiality: confidential
         ? {
             durationYears: parseInt(value.confidentialityDurationYears),
             basis: value.confidentialityBasis.trim()
           }
-        : null
+        : null,
+      ...(value.archiveExternally
+        ? {
+            templateType: 'ARCHIVED_EXTERNALLY',
+            processDefinitionNumber: value.processDefinitionNumber.trim(),
+            archiveDurationMonths: parseInt(value.archiveDurationMonths)
+          }
+        : {
+            templateType: 'REGULAR',
+            processDefinitionNumber: archived
```

Comment:
Ehkä tän archived booleanin vois nimetä uusiks kun se et sillä on TOS numero ei oikeastaan tarkoita että se arkistoitais vaan että se kuuluu asianhallintaprosessien piiriin. Mut se voi kyllä olla ihan oma lappunsa kun pitäis muutenkin nimee uusiks kaikkee tietokantatauluista lähtien. Esim case_management_process tjsp eikä archive_process.

Reply:
Tarkoitatko `archiveExternally`-flägin uudelleen nimeämistä? Mikä sille olisi kuvaavampi nimi?

Reply:
en, se on hyvä nimi, tos on vaan toi toinen `archived` muuttuja joka menee helposti sekaisin sen kanssa, mutta joo tehdään renamet muussa yhteydessä